check_python_format

```diff
@@ -0,0 +1,96 @@
+#! /bin/bash
+
+#
+# A simple script for verifying that all the commits in the PR-range has
+# the code correctly formatted with the Black code formatter.
+#
+set -e
+
+CODE_DIRECTORIES="."
+
+while [[ $# -gt 0 ]]
+do
+    case "$1" in
+        -h|--help)
+            echo "usage: $(basename $0) [OPTIONS] <git-range>"
+            echo ""
+            echo "-d|--directory"
+            echo "  Adds the directory to the directories checked by Black"
+            echo ""
+            echo "This script runs the black formatter on all the python code"
+            echo "it can find, according to the local .pyproject.toml file"
+            echo "and exits with exit-code 1, if any of the commits contains"
+            echo "code which is not formatted in accordance with Black's rules."
+            echo ""
+            exit 0
+            ;;
+        -d|--directory)
+            echo "Adding directory to be verified: $2"
+            CODE_DIRECTORIES="${CODE_DIRECTORIES} $2"
+            shift 2
+            ;;
+        *)
+            break
+            ;;
+    esac
+done
+
+if [ -z "$COMMIT_RANGE" ] && [ -n "$CI_COMMIT_REF_NAME" ]
+then
+    # Gitlab unfortunately doesn't record base branches of commits when the PR
+    # comes from Github, so we need to detect branch names of PRs manually, and
+    # then reconstruct the correct range from that, by excluding all other
+    # branches.
+    case "$CI_COMMIT_REF_NAME" in
+        pr_[0-9]*)
+            EXCLUDE_LIST=$(mktemp)
+            EXCLUDE_LIST_REMOVE=$(mktemp)
+            git for-each-ref --format='%(refname)' | sort > $EXCLUDE_LIST
+            git for-each-ref --format='%(refname)' --points-at $CI_COMMIT_REF_NAME | sort > $EXCLUDE_LIST_REMOVE
+            TO_EXCLUDE="$(comm -23 $EXCLUDE_LIST $EXCLUDE_LIST_REMOVE | tr '\n' ' ')"
+            COMMIT_RANGE="$CI_COMMIT_REF_NAME --not $TO_EXCLUDE"
+            rm -f $EXCLUDE_LIST $EXCLUDE_LIST_REMOVE
+            ;;
+    esac
+fi
+
+if [ -z "$COMMIT_RANGE" ] && [ -n "$TRAVIS_BRANCH" ]
+then
+    COMMIT_RANGE="$TRAVIS_BRANCH..HEAD"
+fi
+
+if [ -z "$COMMIT_RANGE" ]
+then
+    # Just check previous commit if nothing else is specified.
+    COMMIT_RANGE=HEAD~1..HEAD
+fi
+
+if [ -n "$1" ]
+then
+    echo "Checking range: $@:"
+    git --no-pager log "$@"
+    commits="$(git rev-list --no-merges "$@")"
+else
+    echo "Checking range: ${COMMIT_RANGE}:"
+    git --no-pager log $COMMIT_RANGE
+    commits="$(git rev-list --no-merges $COMMIT_RANGE)"
+fi
```

Comment:
Up until here the logic is the same as the check commits script. I wonder if we could move it to a common script with a function that then both could source... ;)

Reply:
he he, yup. It is in the pipes ;)